scene/animation/animation_node_state_machine.cpp

```diff
@@ -331,6 +338,11 @@ double AnimationNodeStateMachinePlayback::process(AnimationNodeStateMachine *p_s
 			// teleport to start
 			if (p_state_machine->states.has(start_request)) {
 				path.clear();
+				if (current != StringName()){
```

Comment:
I am wondering if we should have
state_enter, state_changed, state_exit be emitted exclusively and without overlap from one to the other, so the order would be

* state_enter
* state_changed
* state_exit

Reply:
"exclusively and without overlap" meaning:

- _state_enter_ is only emitted when entering into the machine for the first time, on the first state entered
- _state_changed_ is emitted any time a state transitions into another valid state, automatically or using `Start` or `Travel`
- _state_exit_ is only emitted when a machine stops

or something else?